tests/fe/templates_test.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import os
+from dataclasses import fields, is_dataclass
+from typing import TYPE_CHECKING
+
+from jinja2.meta import find_undeclared_variables
+
+import grouper.fe
+from grouper.fe.settings import FrontendSettings
+from grouper.fe.templates import BaseTemplate
+from grouper.fe.templating import FrontendTemplateEngine
+
+if TYPE_CHECKING:
+    from typing import Set
+
+# These fields are automatically added to the template namespace for all render invocations in
+# BaseTemplate or BaseTemplateEngine and thus don't need to be provided by the template dataclass.
+DEFAULT_FIELDS = {"ROLES", "alerts", "is_active", "static_url", "update_qs", "xsrf_form"}
```

Comment:
I think there's a potential failure case that wouldn't be caught: if one of these is no longer set but a template requires it, the test will still pass. Maybe pull out into constants (lower the chance one of these is randomly no longer set) or pull out the default namespace construction in `BaseTemplate` and `BaseTemplateEngine` into explicitly testable functions?
Alternatively, that failure case might just be unlikely enough that we don't care :)

Separately, should `"TYPES"` also be in here? https://github.com/dropbox/merou/blob/cc70dd41850884e1fc112125dfc693c2458f4f73/grouper/templating.py#L49

Reply:
Will add `TYPES`. I thought about breaking this up into separate testable functions, but the problem I ran into is that `grouper.fe.templating` wants a `GrouperHandler`, which are really difficult to instantiate in a test suite. I think this will work for now; we've almost never changed those variables, and hopefully anyone who does will do a grep to see what else uses them.